### PR TITLE
CMake: folly-deps: gflags is gflags-shared on Fedora #753

### PR DESCRIPTION
**Pull request by kornrunner**
This targets Fedora specifically. More details are available in #753

I personally do not think this is the best way to handle this, but it's something (and works). Still, there might be a chance for this to get merged in? Thank you!

/cc @yfeldblum

### CODE REVIEW

File: CMake/folly-deps.cmake [hunk 1]
```diff
@@ -26,7 +26,11 @@ find_package(gflags CONFIG QUIET)
 if (gflags_FOUND)
   message(STATUS "Found gflags from package config")
   set(FOLLY_HAVE_LIBGFLAGS ON)
-  set(FOLLY_SHINY_DEPENDENCIES ${FOLLY_SHINY_DEPENDENCIES} gflags)
+  if(EXISTS "/etc/fedora-release")
```
Comment (JoeLoser) [inline CMake/folly-deps.cmake]: Is it like this for every Fedora version or only after some threshold?

File: CMake/folly-deps.cmake [hunk 2]
```diff
@@ -26,7 +26,11 @@ find_package(gflags CONFIG QUIET)
...
+    set(FOLLY_SHINY_DEPENDENCIES ${FOLLY_SHINY_DEPENDENCIES} gflags-shared)
```
Comment (JoeLoser) [inline CMake/folly-deps.cmake]: Is it called `gflags-shared` for every Fedora version or only after some threshold?

Comment (kornrunner) [reply CMake/folly-deps.cmake]: I've checked latest gflags packages for:
Fedora Core 27: gflags-2.1.2-5
Fedora Core 26: gflags-2.1.2-2
Fedora Core 25: gflags-2.1.2-1
In all of them it's called `gflags-shared`.

I haven't checked previous versions of packages or distributions. Current stable is FC27. Hope this helps. Thank you!

Comment (Orvid) [reply CMake/folly-deps.cmake]: `gflags-shared` is one of the targets GFlags can choose to export by default. This check can be done as 
```CMake
if (TARGET gflags-shared)
  set(FOLLY_SHINY_DEPENDENCIES ${FOLLY_SHINY_DEPENDENCIES} gflags-shared)
elseif (TARGET gflags)
  set(FOLLY_SHINY_DEPENDENCIES ${FOLLY_SHINY_DEPENDENCIES} gflags)
else()
  message(FATAL_ERROR "Unable to determine the target name for the GFlags package.")
endif()
```

Comment (kornrunner) [reply CMake/folly-deps.cmake]: Awesome! That kind of check was something that I was actually looking for.

Comment (yfeldblum) [reply CMake/folly-deps.cmake]: Please do that.

**Comment by kornrunner** [comment]
Updated PR

**Comment by kornrunner** [comment]
Can we perhaps consider reverting this change? This has worked for Fedora 27 and earlier, with gflags 2.1.2 (and earlier).

It seems that newer editions of Fedora have changed this again (see https://github.com/facebookincubator/fizz/issues/20)

I'm not sure if we wish to keep BC or not, but current state is a show-stopper for newer versions.

Thank you!

**Pull request closed without merging**